editors/sc-ide/core/main.cpp

```diff
@@ -162,8 +163,9 @@ bool SingleInstanceGuard::tryConnect(QStringList const & arguments)
     mIpcServer = new QTcpServer(this);
     bool listening = mIpcServer->listen(QHostAddress(QHostAddress::LocalHost), SingleInstanceGuard::Port);
     if (listening) {
-        mIpcChannel = new ScIpcChannel(mIpcServer->nextPendingConnection(), QString("SingleInstanceGuard"), this);
-
+		// socket is NULL here in Windows, so we pass it in onNewIpcConnection()
+        QTcpSocket *socket = mIpcServer->nextPendingConnection();
+        mIpcChannel = new ScIpcChannel(socket, QString("SingleInstanceGuard"), this);
```

Comment:
Yes, this was to expose the fact that socket is NULL at this point. Might be useful while debugging. I can undo it if you want.